java/google/registry/rde/XjcToDomainResourceConverter.java

```diff
@@ -0,0 +1,329 @@
+import google.registry.model.contact.ContactResource;
+import google.registry.model.domain.DesignatedContact;
+import google.registry.model.domain.DomainResource;
+import google.registry.model.domain.GracePeriod;
+import google.registry.model.domain.rgp.GracePeriodStatus;
+import google.registry.model.domain.secdns.DelegationSignerData;
+import google.registry.model.eppcommon.StatusValue;
+import google.registry.model.host.HostResource;
+import google.registry.model.index.ForeignKeyIndex;
+import google.registry.model.poll.PollMessage;
+import google.registry.model.registry.Registries;
+import google.registry.model.registry.Registry;
+import google.registry.model.reporting.HistoryEntry;
+import google.registry.model.transfer.TransferData;
+import google.registry.model.transfer.TransferStatus;
+import google.registry.util.XmlToEnumMapper;
+import google.registry.xjc.domain.XjcDomainContactType;
+import google.registry.xjc.domain.XjcDomainNsType;
+import google.registry.xjc.domain.XjcDomainStatusType;
+import google.registry.xjc.rdedomain.XjcRdeDomain;
+import google.registry.xjc.rdedomain.XjcRdeDomainElement;
+import google.registry.xjc.rdedomain.XjcRdeDomainTransferDataType;
+import google.registry.xjc.rgp.XjcRgpStatusType;
+import google.registry.xjc.secdns.XjcSecdnsDsDataType;
+import org.joda.time.DateTime;
+
+/** Utility class that converts an {@link XjcRdeDomainElement} into a {@link DomainResource}. */
+final class XjcToDomainResourceConverter extends XjcToEppResourceConverter {
+
+  private static final XmlToEnumMapper<TransferStatus> TRANSFER_STATUS_MAPPER =
+      XmlToEnumMapper.create(TransferStatus.values());
+
+  private static final Function<XjcDomainStatusType, StatusValue> STATUS_CONVERTER =
+      new Function<XjcDomainStatusType, StatusValue>() {
+    @Override
+    public StatusValue apply(XjcDomainStatusType status) {
+      return convertStatusType(status);
+    }
+  };
+
+  private static final Function<String, Key<HostResource>> HOST_OBJ_CONVERTER =
+      new Function<String, Key<HostResource>>() {
+        @Override
+        public Key<HostResource> apply(String fullyQualifiedHostName) {
+          Key<HostResource> key = ForeignKeyIndex.loadAndGetKey(
+              HostResource.class,
+              fullyQualifiedHostName,
+              DateTime.now());
+          checkState(key != null,
+              String.format("HostResource not found with name '%s'", fullyQualifiedHostName));
+          return key;
+        }
+  };
+
+  private static final Function<XjcDomainContactType, DesignatedContact> CONTACT_CONVERTER =
+      new Function<XjcDomainContactType, DesignatedContact>() {
+        @Override
+        public DesignatedContact apply(XjcDomainContactType contact) {
+          return convertContactType(contact);
+        }
+  };
+
+  private static final Function<XjcSecdnsDsDataType, DelegationSignerData> SECDNS_CONVERTER =
+      new Function<XjcSecdnsDsDataType, DelegationSignerData>() {
+
```

Comment:
You have two extra newlines in this one that can be removed for consistency with the previous two.